File: JHU_Scheduler.cpp
#include <Arduino.h>
#include <stdio.h>
#include <string.h>
#include <time.h>

class Employee
{
	public:
	int ID;
	string Name;
	bool CanMoonlight;
	int WeeklyHourReq;
	int ShiftCount;
}

Class Schedule
{
	public:
	int ScheduleID;
	time_t StartDate;
	time_t EndDate;
}

Class Event
{
	private:
	int TypeKey;
	string TypeDescr;

	public:
	int EventID;
	time_t StartTime;
	time_t EndTime
	string EmployeeName;
	string Description
	int Priority
}

Class Location
{
	public:
	string Name;
}

class ShiftOpen
{
	public:
	time_t StartTime;
	time_t EndTime;
}


int main(int argc, const char* argv[])
{
	printf("Welcome to Scheduler!\n");

	return 0;
}

File: JHU_Scheduler.java
package JHU_Scheduler;

import com.opencsv.CSVReader;
import java.io.IOException;
import java.io.Writer;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.*;
import java.text.*;
import java.io.*;

public class Employee {
  public int ID;
	public String Name;
	public boolean CanMoonlight;
	public int WeeklyHourReq;
	public int ShiftCount;
}



public class Schedule {
  public int ScheduleID;

  //NEED TO ADD DATES HERE OR IN THE "Event" CLASS
  private long StartDate;
  private long EndDate;

  

}



public class Event {
  /*
    Private variables and their member functions:
  */
  private int TypeKey;
  private String TypeDescr;

  //Getters:
  int GetTypeKey() {
    return TypeKey;
  }
  String GetTypeDescr() {
    return TypeDescr;
  }

  //Setters:
  void SetTypeKey(int newKey) {
    TypeKey = newKey;
  }
  void SetTypeDecr(String newDescr) {
    TypeDescr = newDescr;
  }

  /*
    Public variables:
  */
  public int eventID;
  //NEED TIMES HERE, BOTH START AND END
  public String EmployeeName;
  public String Description;
  public int Priority;

}



public class Location {
  public String Name;
}



public class ShiftOpen {
  //Times, again...
}
